### PR TITLE
Add dual-stack e2e tests for vsphere

### PR DESCRIPTION
**Pull request by rastislavs**
Signed-off-by: Rastislav Szabo <rastislav@kubermatic.com>

**What does this PR do / Why do we need it**:
Adds dual-stack e2e tests for vsphere.

**Does this PR close any issues?**:
Fixes #

**Special notes for your reviewer**:

**Documentation**:
Just e2e test, no new functionality.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/test/dualstack/cloud_providers.go [hunk 1]
```diff
@@ -397,6 +397,35 @@ func (a equinix) CloudSpec() models.CloudSpec {
...
+			DiskSizeGB: 10,
+		},
+	}
+}
+
+func (a vsphere) CloudSpec() models.CloudSpec {
+	return models.CloudSpec{
+		DatacenterName: "vsphere-ger",
+		Vsphere: &models.VSphereCloudSpec{
+			Username: "VSPHERE_USERNAME",
```
Comment (PratikDeoghare) [inline pkg/test/dualstack/cloud_providers.go]: did you mean `os.Getenv("VSPHERE_USERNAME")`?

Comment (rastislavs) [reply pkg/test/dualstack/cloud_providers.go]: OFC :) thanks

**Review by PratikDeoghare — APPROVED** [review]

**Pull request merged**